README.md

```diff
@@ -0,0 +1,68 @@
+# db-replica
+
+[![license][license img]](LICENSE) [![PRs Welcome][contrib img]](CONTRIBUTING.md)
+
+[license img]: https://img.shields.io/badge/license-Apache%202.0-blue.svg?style=flat-square
+[contrib img]: https://img.shields.io/badge/PRs-welcome-brightgreen.svg?style=flat-square
+
+Using database replicas unlocks horizontal scalability. Some replica designs force replicas to be read-only.
+Read-only queries can be sent to the replica, while others have to go to the main DB.
+The `db-replica` API automatically routes the queries to the correct node.
+It integrates at the `java.sql.Connection` level, so you don't have to hunt down hundreds of queries manually.
+
+## Usage
+
+```java
+import com.atlassian.db.replica.api.*;
+import java.sql.*;
+import java.time.*;
+
+class Example {
+
+    private final ReplicaConsistency consistency = ReplicaConsistency.assumePropagationDelay(
+        Duration.ofMillis(100),
+        Clock.systemUTC(),
+        Cache.cacheMonotonicValuesInMemory()
+    );
+
+    ResultSet queryReplicaOrMain(String sql) {
+        try (ConnectionProvider connectionProvider = new PostgresConnectionProvider()) {
+            Connection connection = DualConnection.builder(connectionProvider, consistency).build();
+            return connection.prepareStatement(sql).executeQuery();
+        }
+    }
+}
+```
+
+## Installation
+
+Maven:
+```xml
+<dependency>
+    <groupId>com.atlassian.db.replica</groupId>
+    <artifactId>db-replica</artifactId>
+    <version>0.1.17</version>
+</dependency>
+```
+
+## Documentation
+
+TODO: make Javadocs great
```

Comment:
@dagguh Would it be possible to have some minimum documentation before the checklist PR get's merged?

Reply:
@dagguh I think we can link dual-connection-states.md here.

Reply:
@highvoltag3 roger that, done!
@wyrzyk yes indeed ❤️